src/submission/submission-service.ts

```diff
@@ -892,25 +896,59 @@ export namespace operations {
           .processParallel(command.clinicalType, record, index, schema);
 
         if (schemaResult.validationErrors.length > 0) {
-          let validationErrors = schemaResult.validationErrors;
+          let validationErrors: DeepReadonly<SchemaValidationError>[] = [
+            ...schemaResult.validationErrors,
+          ];
+
           if (FEATURE_SUBMISSION_EXCEPTIONS_ENABLED) {
+            // check for program exception
             const result = await programExceptionRepository.find(command.programId);
 
-            // filter out valid exceptions before adding to error accumulator
-            validationErrors =
-              // only filter is we have valid exceptions available
-              isProgramException(result)
-                ? schemaResult.validationErrors.filter(validationError => {
-                    return !applyExceptionIfExists(result.exceptions, validationError, record);
-                  })
-                : schemaResult.validationErrors;
+            validationErrors = [];
+
+            if (isProgramException(result)) {
+              schemaResult.validationErrors.forEach(validationError => {
+                const validationErrorFieldName = validationError.fieldName;
+                const recordValue = record[validationErrorFieldName];
+
+                // Zero Array type exceptions exist, but recordValue type is string | string[]
+                if (Array.isArray(recordValue)) {
+                  validationErrors.push(validationError);
+                  return;
+                }
+
+                const normalizedRecordValue = normalizeExceptionValue(recordValue);
+
+                const exceptionExists = checkExceptionExists(
+                  result.exceptions,
+                  validationError,
+                  normalizedRecordValue,
+                );
+
+                if (exceptionExists) {
+                  // ensure value is normalized exception value
+                  const normalizedExceptionRecord = {
+                    ...record,
+                    ...{ [validationErrorFieldName]: normalizedRecordValue },
+                  };
+                  processedRecord = normalizedExceptionRecord;
+                } else {
+                  // only add validation errors that don't have exceptions
+                  validationErrors.push(validationError);
```

Comment:
I like the decision to do this but it is kind of subtle... You are normalizing the input to exactly match the normalized exception value. This deserves to be stated in a comment that we are doing this, probably near the top of this block or top of the method. Somethign should say that "Not only are we allowing the value through if there is an applicable exception, but we will normalize the value to exactly match the allowed value. Normalizing is setting it to start Upper case and to trim whitespace"... something like that

something